akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -177,6 +177,20 @@ object Source {
     new Source(scaladsl.Source.fromCompletionStage(future))
 
   /**
+   * Starts a new `Source` from another `future` source.
+   * The stream will consist of the elements of the given source,
+   * once it successfully completes.
+   */
+  def fromFutureSource[T, M](future: Future[Graph[SourceShape[T], M]]): javadsl.Source[T, Future[M]] = new Source(scaladsl.Source.fromFutureSource(future))
```

Comment:
strange that we provide methods for both `Future` and `CompletionStage`, but I can see that we have that in other places also